src/coreclr/tools/Common/Compiler/ProcessLinkerXmlBase.cs

```diff
@@ -217,9 +217,10 @@ protected virtual void ProcessTypes(ModuleDesc assembly, XPathNavigator nav, boo
                         continue;
                 }
 
-                // TODO: Process exported types
+                MetadataType? type = CecilCompatibleTypeParser.GetType(assembly, fullname);
 
-                TypeDesc? type = CecilCompatibleTypeParser.GetType(assembly, fullname);
```

Comment:
This was already returning a resolved type, so exported types worked, but we "forgot" we got to the type through a forwarder and didn't preserve it. The newly added `type.Module != assembly` is basically "was this a forwarder?".